python/python/Ice/Util.py

```diff
@@ -70,23 +70,8 @@ def initialize(
             await greeter.greetAsync()
     """
 
-    if args and not isinstance(args, list):
-        raise InitializationException("args must be a list of strings")
-
-    if initData and not isinstance(initData, InitializationData):
-        raise InitializationException("initData must be an instance of Ice.InitializationData")
-
-    if initData and args:
-        raise InitializationException("Both args and initData arguments cannot be specified")
-
-    if initData and eventLoop:
-        raise InitializationException("Both initData and eventLoop arguments cannot be specified")
-
     eventLoopAdapter = initData.eventLoopAdapter if initData else None
     if eventLoop:
-        assert not eventLoopAdapter  # initData is None here
```

Comment:
I think it's clearer for the reader to keep this assert. Otherwise, you easily get the impression you assign eventLoopAdapter on line 73 and then blindly reassign it a few lines below.

Reply:
maybe is even clear to write:
```
    eventLoopAdapter = None
    if initData:
        eventLoopAdapter = initData.eventLoopAdapter
    elif eventLoop:
        eventLoopAdapter = AsyncIOEventLoopAdapter(eventLoop)
```